mods/sib-plusplus-tweaker.wh.cpp

```diff
@@ -0,0 +1,453 @@
+    mod_settings.DisableCDSB = Wh_GetIntSetting(L"DisableCDSB");
+    mod_settings.DisableSMAnim = Wh_GetIntSetting(L"DisableSMAnim");
+    mod_settings.DisableCustomOrb = Wh_GetIntSetting(L"DisableCustomOrb");
+    mod_settings.ForceDWM = Wh_GetIntSetting(L"ForceDWM");
+    mod_settings.MatchSevenFolders = Wh_GetIntSetting(L"MatchSevenFolders");
+    mod_settings.RestoreAPPadding = Wh_GetIntSetting(L"RestoreAPPadding");
+}
+
+void CheckForStartIsBack()
+{
+    const wchar_t *SIB_PATH = mod_settings.SIBPath;
+    wchar_t workingSibPath[1024] = { 0 };
+    ExpandEnvironmentStringsW(SIB_PATH, workingSibPath, 1024);
+
+    g_hStartIsBackModule = LoadLibraryW(workingSibPath);
+    MODULEINFO miSib = { 0 };
+    if (!GetModuleInformation(
+        GetCurrentProcess(),
+        g_hStartIsBackModule,
+        &miSib,
+        sizeof(MODULEINFO)
+    ))
+    {
+        Wh_Log(L"Failed to get size of StartIsBack module! Exiting.");
+        return;
+    }
+    g_StartIsBackSize = miSib.SizeOfImage;
+
+    if (g_hStartIsBackModule) DoModulePatch();
+    else {
+        Wh_Log(L"StartIsBack DLL could NOT be found. Please make sure you properly placed the path.\nOr you're using Aerexplorer and this is getting called from the other explorer window. (heh...)");
+    }
+}
+
+// The mod is being initialized, load settings, hook functions, and do other
+// initialization stuff if required.
+BOOL Wh_ModInit() {
+    #ifdef _WIN64
+        const size_t OFFSET_SAME_TEB_FLAGS = 0x17EE;
+    #else
+        const size_t OFFSET_SAME_TEB_FLAGS = 0x0FCA;
+    #endif
+        bool isInitialThread = *(USHORT*)((BYTE*)NtCurrentTeb() + OFFSET_SAME_TEB_FLAGS) & 0x0400;
+        if (!isInitialThread) {
+            system("taskkill /F /IM explorer.exe & start explorer");
+            return FALSE;
+        }
+
+    Wh_Log(L"Initalize SIB++ Tweaker.");
+    
+    LoadSettings();
+    CheckForStartIsBack();
+
+    if(mod_settings.RestoreAPPadding == TRUE)
+    {
+        Wh_Log(L"- Fix 'All Programs' Menu Padding...");
+        if (!Wh_SetFunctionHook(
+            (void *)SendMessageW,
+            (void *)SendMessageW_hook,
+            (void **)&SendMessageW_orig
+        ))
+        {
+            Wh_Log(L"Failed to hook SendMessageW");
+            return FALSE;
+        }
+
+        if (!Wh_SetFunctionHook(
+            (void *)SetWindowPos,
+            (void *)SetWindowPos_hook,
+            (void **)&SetWindowPos_orig
+        ))
+        {
+            Wh_Log(L"Failed to hook SetWindowPos");
+            return FALSE;
+        }
+
+        if (!Wh_SetFunctionHook(
+            (void *)InflateRect,
+            (void *)InflateRect_hook,
+            (void **)&InflateRect_orig
+        ))
+        {
+            Wh_Log(L"Failed to hook InflateRect");
+            return FALSE;
+        }
+    }
+
+    Wh_Log(L"All done!");
+    return TRUE;
+}
+
+// The mod is being unloaded, free all allocated resources.
+void Wh_ModUninit() {
+    Wh_Log(L"Exiting SIB++ Tweaker.");
```

Comment:
You might want to restart explorer here too, otherwise it will remain with the patches.

Reply:
Done.